at-dungnguyen/Final/main.rb

```diff
@@ -0,0 +1,399 @@
+    p '6. Delete a notification' if @user['role'] == SessionController::LEADER
+    p '0. Exit'
+  end
+
+  def execute_notify(function)
+    case function
+    when 1
+      all_notify
+    when 2
+      g_notify
+    when 3
+      @flag_notify = false
+    when 4
+      @flag_notify = false
+      @user = nil
+    when 5
+      send_notify
+    when 6
+      del_notify
+    when 0
+      @flag_notify = false
+      execute2(0)
+    end
+  end
+
+  def all_notify
+    if @user['team']
+      list_notify = @nc.get_items(@user['team'])
+      if list_notify.any?
+        list_notify.each do |notify|
+          puts "#{notify['id']} : #{notify['title']} | #{notify['publish_date']}"
+        end
+      end
+    end
+    p '====================================================================='
+  end
+
+  def g_notify
+    p 'Input id notification'
+    id = input
+    notify = @nc.get_item(id, @user)
+    clear
+    if notify
+      p "#{notify['id']} : #{notify['title']} | #{notify['publish_date']}"
+      p notify['content']
+    else
+      p 'Invalid ID'
+    end
+    p '================================='
+  end
+
+  def send_notify
+    p 'Input title'
+    title = gets.chomp
+    p 'Input content'
+    content = gets.chomp
+    publish_date = Time.now
+    id_team = @user['team']
+    notify = Notification.new(title, content, publish_date.strftime('%F %T'), id_team)
+    clear
+    if @nc.add_item(notify)
+      p 'Send notify success'
+    else
+      p 'Send error!'
+    end
+    p '================================'
+  end
+
+  def del_notify
+    p 'Input id notification'
+    id = input
+    clear
+    if @nc.del_item(id)
+      p 'Delete Notification Success'
+    else
+      p "Notification doesn't exist!"
+    end
+    p '================================'
+  end
+
+  # Leave
+  def leave_management
+    p 'Leave Management'
+    p '==============='
+    p '1. Send a leave request'
+    p '2. Update a leave request'
+    p '3. Delete a leave request'
+    p '4. Show list of leaves of employees.'
+    p '5. Back'
+    p '6. Logout'
+    p '7. Show list leaves request need approve' if @user['role'] == SessionController::LEADER
+    p '8. Approve or reject leave request' if @user['role'] == SessionController::LEADER
+    p '0. Exit'
+  end
+
+  def execute_leave(function)
+    case function
+    when 1
+      send_request
+    when 2
+      update_request
+    when 3
+      delele_request(@user)
+    when 4
+      list_approved
+    when 5
+      @flag_leave = false
+    when 6
+      @flag_leave = false
+      @user = nil
+    when 7
+      list_need_approve if @user['role'] == SessionController::LEADER
+    when 8
+      update_status if @user['role'] == SessionController::LEADER
+    when 0
+      @flag_leave = false
+      execute2(0)
+    end
+  end
+
+  def send_request
+    p 'Reason'
+    reason = gets.chomp
+    p 'Total date off'
+    total_date_off = input
+    time = Time.now
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/atruby-team/internship-18-summer/blob/3c0bf6c6d3420a006a4c5cda03d7111306060898/at-dungnguyen/Final/main.rb#L250"></span>Rails/TimeZone: Do not use <code>Time.now</code> without zone. Use one of <code>Time.zone.now</code>, <code>Time.current</code>, <code>Time.now.in_time_zone</code>, <code>Time.now.utc</code>, <code>Time.now.getlocal</code>, <code>Time.now.xmlschema</code>, <code>Time.now.iso8601</code>, <code>Time.now.jisx0301</code>, <code>Time.now.rfc3339</code>, <code>Time.now.httpdate</code>, <code>Time.now.to_i</code>, <code>Time.now.to_f</code> instead. (https://github.com/rubocop-hq/rails-style-guide#time, http://danilenko.org/2012/7/6/rails_timezones)</td>
    </tr>
  </tbody>
</table>